### PR TITLE
[Improvement]: Refactor TableFormat to a non-enumeration type to support non-open source formats

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

We are trying to adapt a non-standard tableFormat using Amoro

## Brief change log

- Refactor TableFormat to a non-enum type.
- Replace all `==` comparisons in your code with `Object.equals`

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-ams/src/main/java/org/apache/amoro/server/persistence/mapper/TableMetaMapper.java [hunk 1]
```diff
@@ -244,7 +257,7 @@ TableMetadata selectTableMetaByName(
   @Insert(
       "INSERT INTO table_identifier(catalog_name, db_name, table_name, format) VALUES("
           + " #{tableIdentifier.catalog}, #{tableIdentifier.database}, #{tableIdentifier.tableName}, "
-          + " #{tableIdentifier.format})")
+          + " #{tableIdentifier.format, typeHandler=org.apache.amoro.server.persistence.converter.TableFormatConverter})")
```
Comment (zhoujinsong) [inline amoro-ams/src/main/java/org/apache/amoro/server/persistence/mapper/TableMetaMapper.java:260]: It seems the full class name is not necessary here.

File: amoro-format-iceberg/src/test/java/org/apache/amoro/catalog/TableTestBase.java [hunk 2]
```diff
@@ -44,14 +45,11 @@ public void setupTable() {
...
-    switch (getTestFormat()) {
-      case MIXED_HIVE:
-      case MIXED_ICEBERG:
-        createMixedFormatTable();
-        break;
-      case ICEBERG:
-        createIcebergFormatTable();
-        break;
+    TableFormat format = getTestFormat();
+    if (TableFormat.MIXED_HIVE.equals(format) || TableFormat.MIXED_ICEBERG.equals(format)) {
```
Comment (zhoujinsong) [inline amoro-format-iceberg/src/test/java/org/apache/amoro/catalog/TableTestBase.java:49]: How about replacing with TableFormat#in?

File: amoro-format-iceberg/src/main/java/org/apache/amoro/mixed/CatalogLoader.java [hunk 3]
```diff
@@ -88,7 +88,7 @@ private static String catalogImpl(String metastoreType, Map<String, String> cata
         tableFormats.size() == 1, "Catalog support only one table format now.");
     TableFormat tableFormat = tableFormats.iterator().next();
     Preconditions.checkArgument(
-        TableFormat.MIXED_HIVE == tableFormat || TableFormat.MIXED_ICEBERG == tableFormat,
+        TableFormat.MIXED_HIVE.equals(tableFormat) || TableFormat.MIXED_ICEBERG.equals(tableFormat),
```
Comment (zhoujinsong) [inline amoro-format-iceberg/src/main/java/org/apache/amoro/mixed/CatalogLoader.java:91]: How about replacing with TableFormat#in?

File: amoro-ams/src/main/java/org/apache/amoro/server/optimizing/maintainer/TableMaintainer.java [hunk 4]
```diff
@@ -60,9 +60,9 @@ default void cleanDanglingDeleteFiles(TableRuntime tableRuntime) {
 
   static TableMaintainer ofTable(AmoroTable<?> amoroTable) {
     TableFormat format = amoroTable.format();
-    if (format == TableFormat.MIXED_HIVE || format == TableFormat.MIXED_ICEBERG) {
+    if (TableFormat.MIXED_HIVE.equals(format) || TableFormat.MIXED_ICEBERG.equals(format)) {
```
Comment (zhoujinsong) [inline amoro-ams/src/main/java/org/apache/amoro/server/optimizing/maintainer/TableMaintainer.java:63]: How about replacing with TableFormat#in?

**Review by zhoujinsong — COMMENTED** [review]
@baiyangtx Thanks for the contribution, I left some comments, PTAL.

File: amoro-ams/src/main/java/org/apache/amoro/server/optimizing/maintainer/TableMaintainer.java (hunk 4 above)
Comment (baiyangtx) [reply amoro-ams/src/main/java/org/apache/amoro/server/optimizing/maintainer/TableMaintainer.java]: fixed

File: amoro-format-iceberg/src/test/java/org/apache/amoro/catalog/TableTestBase.java (hunk 2 above)
Comment (baiyangtx) [reply amoro-format-iceberg/src/test/java/org/apache/amoro/catalog/TableTestBase.java]: fixed

File: amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/TableController.java [hunk 5]
```diff
@@ -497,18 +497,14 @@ public void getTableList(Context ctx) {
...
-              return TableMeta.TableType.ICEBERG.toString();
-            case HUDI:
-              return TableMeta.TableType.HUDI.toString();
-            default:
-              throw new IllegalStateException("Unknown format");
+          if (format.equals(TableFormat.MIXED_HIVE) || format.equals(TableFormat.MIXED_ICEBERG)) {
+            return TableMeta.TableType.ARCTIC.toString();
+          } else if (format.equals(TableFormat.PAIMON)) {
+            return TableMeta.TableType.PAIMON.toString();
+          } else if (format.equals(TableFormat.ICEBERG)) {
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply amoro-ams/src/main/java/org/apache/amoro/server/dashboard/controller/TableController.java:504]: fixed

File: amoro-ams/src/main/java/org/apache/amoro/server/persistence/mapper/TableMetaMapper.java (hunk 1 above)
Comment (baiyangtx) [reply amoro-ams/src/main/java/org/apache/amoro/server/persistence/mapper/TableMetaMapper.java:260]: Only fullname can be used here

**Review by zhoujinsong — APPROVED** [review]
LGTM.

Thanks for the contribution!

**Pull request merged**